app/views/homepage/_grid_item.haml

```diff
@@ -1,11 +1,12 @@
-.home-fluid-thumbnail-grid-item
-  %div
-    -# Listing image
-    = render :partial => "layouts/grid_item_listing_image", :locals => {:listing => listing, :modifier_class => ""}
+- cache(listing) do
+  .home-fluid-thumbnail-grid-item
+    %div
+      -# Listing image
+      = render :partial => "layouts/grid_item_listing_image", :locals => {:listing => listing, :modifier_class => ""}
 
-    -# Listing author details
-    .home-fluid-thumbnail-grid-author
-      .home-fluid-thumbnail-grid-author-avatar
-        = small_avatar_thumb(listing.author, :class => "home-fluid-thumbnail-grid-author-avatar-image")
-      = link_to(listing.author, :class => "home-fluid-thumbnail-grid-author-name") do
-        = listing.author.name(@current_community)
+      -# Listing author details
+      .home-fluid-thumbnail-grid-author
+        .home-fluid-thumbnail-grid-author-avatar
+          = small_avatar_thumb(listing.author, :class => "home-fluid-thumbnail-grid-author-avatar-image")
+        = link_to(listing.author, :class => "home-fluid-thumbnail-grid-author-name") do
```

Comment:
After this change, changing usernames is not possible anymore, or it'll break these links. Or at least without invalidating cache.